bundles/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/handler/NikoHomeControlEnergyMeterHandler.java

```diff
@@ -0,0 +1,237 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.nikohomecontrol.internal.handler;
+
+import static org.eclipse.smarthome.core.types.RefreshType.REFRESH;
+import static org.openhab.binding.nikohomecontrol.internal.NikoHomeControlBindingConstants.CHANNEL_POWER;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.measure.quantity.Power;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.nikohomecontrol.internal.protocol.NhcEnergyMeter;
+import org.openhab.binding.nikohomecontrol.internal.protocol.NhcEnergyMeterEvent;
+import org.openhab.binding.nikohomecontrol.internal.protocol.NikoHomeControlCommunication;
+import org.openhab.binding.nikohomecontrol.internal.protocol.nhc2.NhcEnergyMeter2;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link NikoHomeControlEnergyMeterHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Mark Herwege - Initial Contribution
+ */
+@NonNullByDefault
+public class NikoHomeControlEnergyMeterHandler extends BaseThingHandler implements NhcEnergyMeterEvent {
+
+    private final Logger logger = LoggerFactory.getLogger(NikoHomeControlEnergyMeterHandler.class);
+
+    private volatile @NonNullByDefault({}) NhcEnergyMeter nhcEnergyMeter;
```

Comment:
But the field is set in the initialize method. Quoting from the development documentation:
Fields within ThingHandler classes that are initialized within the initialize() method may also be annotated like this, because the framework ensures that initialize() will be called before any other method.
I have this same construct in similar handlers (Action, Thermostat) in the current release version of the binding.